src/Microsoft.AspNet.Hosting/Internal/HostingEngine.cs

```diff
@@ -87,6 +88,7 @@ public virtual IDisposable Start()
                 _applicationLifetime.NotifyStopping();
                 server.Dispose();
                 _applicationLifetime.NotifyStopped();
+                loggerFactory.Dispose();
```

Comment:
Agreed, this is weird. Isn't this what DI scopes are for? Would it make sense to create a scope on line 66 and dispose it here? Then any scoped services would be disposed.